src/core/packages/saved-objects/migration-server-internal/src/document_migrator/validate_migration.test.ts

```diff
@@ -112,75 +112,70 @@ describe('validateTypeMigrations', () => {
       expect(() => validate({ type })).not.toThrow();
     });
 
-    describe('when switchToModelVersionAt is specified', () => {
-      it('throws if a migration is specified for a version superior to switchToModelVersionAt', () => {
+    describe('effect of globalSwitchToModelVersionAt', () => {
+      it('throws if a migration is specified for a version superior to globalSwitchToModelVersionAt', () => {
         const type = createType({
           name: 'foo',
-          switchToModelVersionAt: '8.9.0',
           migrations: {
-            '8.10.0': jest.fn(),
+            '8.11.0': jest.fn(),
           },
         });
 
         expect(() =>
-          validate({ type, kibanaVersion: '8.10.0' })
+          validate({ type, kibanaVersion: '8.11.0' })
         ).toThrowErrorMatchingInlineSnapshot(
-          `"Migration for type foo for version 8.10.0 registered after switchToModelVersionAt (8.9.0)"`
+          `"Migration for type foo for version 8.11.0 registered after globalSwitchToModelVersionAt (8.10.0)"`
```

Comment:
Validation still throws an error, so we should catch situations where new migrations are added.